Server/Controllers/MatchController.cs

```diff
@@ -258,9 +267,20 @@ public async Task<ActionResult> After([FromBody] AfterMatchRequest request)
 
             _context.AddRange(newPlayers);
             _context.Matches.Add(match);
-            await _context.SaveChangesAsync();
 
-            return Ok();
+            var playersKvp = request.Players.ToDictionary(x => x.SteamId, x => x.Team);
+            var teams = _ratingService.SplitTeams(playersKvp, request.Winner, allPlayers);
+            await _ratingService.UpdateNewRating(teams[GameResult.Winner], teams[GameResult.Loser]);
```

Comment:
I'd rather make the service do both of these steps, in a single `RecordRankedMatch` call. It also might return the rating changes, instead of manually keeping a track of it inside the controller, i.e.

```csharp
	public class RatingService
	{
		public Dictionary<long, PlayerRatingChange> RecordRankedMatch(IEnumerable<MatchTeam> teams, ushort winner)
	}

	public class PlayerRatingChange
    {
        public int Old { get; set; }
        public int New { get; set; }
    }
```

```csharp
var ratingChanges = request.CustomGame == CustomGame.Dota12v12 ? _ratingService.RecordRankedMatch(match.Players, request.Winner) : null;

...
return new AfterMatchResponse()
{
    Players = allPlayers
        .Select(p => new AfterMatchResponse.Player()
        {
            SteamId = p.SteamId.ToString(),
            RatingChange = ratingChanges?[p.SteamId],
        })
        .ToList(),
};
```

Reply:
If I would have had a response example in the initial task it would have been easier. I did the changes, although, again, lot's of things are being done in one call which makes it hard for code readability, debugging and adds no value

Reply:
It does isolate the related business logic in a single place, instead of dragging it out to the controller. These operations are never going to be individually used anyway.